src/asset/asset.js

```diff
@@ -464,7 +464,12 @@ class Asset extends EventHandler {
     ready(callback, scope) {
         scope = scope || this;
 
-        if (this.resource) {
+        // If cubemap asset doesn't need to load the faces, then resource is null as it references
+        // resources[0] that stores the faces. We need to check if the prefiltered assets are
+        // loaded instead
+        const assetLoaded = this.resource || (this.type === "cubemap" && !this.loadFaces && this.resources.length > 0);
+
+        if (assetLoaded) {
```

Comment:
Looks like this code was written years ago, perhaps before `loaded` flag was introduced. 

The correct code should be:
```suggestion
        if (this.loaded) {
```
This change would impact all assets though and will need thorough testing, but I think it's worth it.

Reply:
I will have another look at this as I vaguely remember there being an issue where something could be 'loaded' but the resource not be ready but can't remember the details